src/python/pants/engine/target.py

```diff
@@ -1442,6 +1449,9 @@ def rules():
     input: ClassVar[Type[Sources]]
     output: ClassVar[Type[Sources]]
 
+    def debug_hint(self) -> str:
+        return "{self.protocol_target.address.spec}"
```

Comment:
I debated this debug hint being `f"{address} to {self.output.__name__}`, which would look like `('//:foo' to PythonSources)`. But I removed it because I realized that the `@rule` names should already express what language it's converting to.

A concern I have is the stacktrace getting too big and unwieldy.

Reply:
See my comment in #10882: it's possible that being thrifty with the `desc` and then rendering them when `--no-print-exception-stacktrace` as a purely end-user readable explanation is a path forward.